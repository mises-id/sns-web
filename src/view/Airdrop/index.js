/*
 * @Author: lmk
 * @Date: 2021-07-16 00:15:24
 * @LastEditTime: 2022-06-29 10:29:47
 * @LastEditors: lmk
 * @Description: Airdrop page
 */
import { getAirdropInfo, getTwitterAuth } from "@/api/user";
import Navbar from "@/components/NavBar";
import { useRouteState } from "@/utils";
import { Button } from "antd-mobile";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useSelector } from "react-redux";
import { useDidRecover } from "react-router-cache-route";
import { useHistory } from "react-router-dom/cjs/react-router-dom.min";
import { ActivityIndicator } from "zarm";
import "./index.scss";
const Airdrop = () => {
  const { t } = useTranslation();
  const [showSkip, setShowSkip] = useState(false);
  const historyState = useRouteState();
  const [MIS, setMIS] = useState(0);
  const history = useHistory();
  const [pageLoading, setPageLoading] = useState(true);
  useDidRecover(() => {
    window.refreshByCacheKey("/airdrop");
  });
  const init = () => {
    const flag = historyState.isFrom === "homePage";
    setShowSkip(!flag);
    setMIS(historyState.MIS);
    const isRefresh = sessionStorage.getItem("isRefresh");
    if (!flag && !isRefresh) {
      sessionStorage.setItem("isRefresh", true);
      window.location.reload();
    }
    if (flag && MIS === 0) {
      getAirdropInfo()
        .then((res) => {
          if (res.airdrop) {
            let url = null;
            const { status, coin } = res.airdrop;
            switch (status) {
              case "default":
                url = `/airdrop?isFrom=homePage&MIS=${coin}`;
                break;
              case "success":
                url = "/airdropSuccess";
                break;
              case "failed":
                url = "/airdropResult?code=1";
                break;
              default:
                // url="/airdrop?isFrom=homePage"
                break;
            }
            setPageLoading(false);
            url && history.replace(url);
            return;
          }else{
            setPageLoading(false);
          }
        })
        .catch((err) => {});
    } else {
      setPageLoading(false);
    }
  };
  useEffect(() => {
    init();
    return () => {
      setMIS(0);
    };
    // eslint-disable-next-line
  }, []);
  const [loading, setLoading] = useState(false);
  const getAuth = () => {
    setLoading(true);
    getTwitterAuth()
      .then((res) => {
        window.location.href = res.url;
      })
      .catch((err) => {
        console.log(err);
      })
      .finally(() => {
        setLoading(false);
      });
  };
  const skip = () => {
    window.history.back();
  };
  const {loginForm} = useSelector(state=>state.user)
  const inviteFriends = () => {
    history.push(`/myInvitation?misesId=${loginForm.misesid}`);
  };
  return (
    <div>
      <Navbar
        title={t("airdropPageTitle")}
        rightChild={
          showSkip ? (
            <div className="skip" onClick={skip}>
              Skip
            </div>
          ) : (
            ""
          )
        }
      />
      {!pageLoading ? (
        <div className="airdrop-bg m-font18">
          <div className="layout-box">
            <img
              src="/static/images/background.png"
              alt=""
              className="background"
            />
            <div className="layout-content">
              <img
                src="/static/images/title.png"
                alt=""
                className="airdrop-title"
              />
              <div className="toClaim-box">
                <img
                  src="/static/images/parachute.png"
                  alt=""
                  className="airdrop-parachute"
                />
                <div className="toClaim-content">
                  {MIS > 0 ? (
                    <>
                      <p className="claimedMis">{`You have successfully claimed ${MIS}MIS`}</p>
                      <p className="desc-item">
                        Airdropping... This process could take a few minutes.
                        <img
                          src="/static/images/circle.png"
                          alt=""
                          className="circle"
                        />
                      </p>
                      <p className="desc-item">
                        Invite your friends to get more MIS!
                        <img
                          src="/static/images/circle.png"
                          alt=""
                          className="circle"
                        />
                      </p>
                      <div className="sendTweet">
                        <Button
                          fill="solid"
                          type="primary"
                          onClick={inviteFriends}
                          className="airdrop-btn"
                        >
                          <span className="airdrop-btn-txt">Invite Friends</span>
                        </Button>
                      </div>
                    </>
                  ) : (
                    <>
                      <p className="toClaim">{t("toClaim")}:</p>
                      <p className="desc-item">
                        {t("airdropContentdesc")}
                        <img
                          src="/static/images/circle.png"
                          alt=""
                          className="circle"
                        />
                      </p>
                      <p className="desc-item">
                        {t("airdropContentdesc1")}
                        <img
                          src="/static/images/circle.png"
                          alt=""
                          className="circle"
                        />
                      </p>
                      <div className="sendTweet">
                        <Button
                          fill="solid"
                          type="primary"
                          loading={loading}
                          onClick={getAuth}
                          className="airdrop-btn"
                        >
                          <span className="airdrop-btn-txt">
                            {t("sendTweet")}
                          </span>
                        </Button>
                      </div>
                    </>
                  )}

                  <img
                    src="/static/images/ribbon.png"
                    alt=""
                    className="ribbon"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      ) : (
        <div style={{ textAlign: "center", padding: "20px" }}>
          <ActivityIndicator type="spinner" />
        </div>
      )}
    </div>
  );
};
export default Airdrop;
