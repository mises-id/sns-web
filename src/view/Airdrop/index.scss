.airdrop-title{
  width: 86%;
  display: block;
  margin:82px auto 0;
}
.airdrop-parachute{
  width: 70px;
  display: block;
  margin: 0 auto;
  position: relative;
  z-index: 3;
}
.toClaim-content{
  border: 3px solid #BAEBFC;
  border-radius: 15px;
  background: white;
  margin: -40px 15px 0;
  padding: 29px 24px;
  position:relative;
  z-index: 2;
  .ribbon{
    position: absolute;
    width: 34px;
    bottom: 36px;
    left: -15px;
  }
}
.claimedMis{
  font-size: 18px;
  color: #333333;
  font-weight: bold;
  margin: 22px 0;
  line-height: 24px;
  word-break: keep-all;
}
.toClaim{
  font-size: 22px;
  color: #333333;
  font-weight: bold;
  margin-bottom: 25px;
}
.desc-item{
  font-size: 16px;
  color: #333333;
  line-height: 21px;
  margin-bottom: 20px;
  font-weight: 400;
  position: relative;
  word-break:break-word;
  .circle{
    position: absolute;
    left: 0;
    width: 15px;
    height: 15px;
    top: -2px;
    left: -7px;
  }
}
.sendTweet{
  margin: 0 auto 10px;
  padding-top: 10px;
}
.airdrop-btn-txt{
  color: #000000;
  font-size: 17px;
  font-weight: 600;
}
.skip{
  font-size: 12px;
  color: #666666;
  border: 1px solid #DDDDDD;
  border-radius: 50px;
  width: 50px;
  height: 25px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.airdrop-bg{
  background-color: #daf6fa;
}
.background{
  display: block;
  width: 100%;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
}
.layout-box{
  position: relative
}
.layout-content{
  position: relative;
  z-index: 2;
  overflow: hidden;
}
.airdrop-btn{
  height: 50px;
  border-radius: 50px;
  width: 100%;
  background: #FFD72F;
  color:#000;
  font-size: 17px;
  font-weight: bold;
  border: none;
  box-shadow: 0px 8px 15px 0px rgba(255, 215, 47, 0.3);
}