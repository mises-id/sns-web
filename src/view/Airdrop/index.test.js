/*
 * @Author: lmk
 * @Date: 2021-07-16 10:05:53
 * @LastEditTime: 2022-03-03 16:30:29
 * @LastEditors: lmk
 * @Description: Airdrop test
 */

import React from 'react';
import {shallow } from 'enzyme';
import Airdrop from './';
describe('GreatePosts', () => {
  it('should render correctly', () => {
    const component = shallow(< Airdrop/>)
    expect(component).toMatchSnapshot();
  });
});