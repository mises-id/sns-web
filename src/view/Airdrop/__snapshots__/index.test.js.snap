// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreatePosts should render correctly 1`] = `
<div>
  <Navbar
    rightChild=""
    title="airdropPageTitle"
  />
  <div
    className="airdrop-bg m-font18"
  >
    <div
      className="layout-box"
    >
      <img
        alt=""
        className="background"
        src="/static/images/background.png"
      />
      <div
        className="layout-content"
      >
        <img
          alt=""
          className="airdrop-title"
          src="/static/images/title.png"
        />
        <div
          className="toClaim-box"
        >
          <img
            alt=""
            className="airdrop-parachute"
            src="/static/images/parachute.png"
          />
          <div
            className="toClaim-content"
          >
            <p
              className="toClaim"
            >
              toClaim
              :
            </p>
            <p
              className="desc-item"
            >
              airdropContentdesc
              <img
                alt=""
                className="circle"
                src="/static/images/circle.png"
              />
            </p>
            <p
              className="desc-item"
            >
              airdropContentdesc1
              <img
                alt=""
                className="circle"
                src="/static/images/circle.png"
              />
            </p>
            <div
              className="sendTweet"
            >
              <ForwardRef
                className="airdrop-btn"
                fill="solid"
                loading={false}
                onClick={[Function]}
                type="primary"
              >
                <span
                  className="airdrop-btn-txt"
                >
                  sendTweet
                </span>
              </ForwardRef>
            </div>
            <img
              alt=""
              className="ribbon"
              src="/static/images/ribbon.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
